lib/resty/cassandra/cluster.lua

```diff
@@ -550,6 +551,15 @@ function _Cluster:refresh()
   -- initiate the load balancing policy
   self.lb_policy:init(peers)
 
+  if self.stream_ids == nil then
+    -- Initialize the list of available seed ids (only once)
+    self.stream_ids = {}
```

Comment:
I agree that using the list implementation provides more features than using the counter.
I'm just concerned about the performance impact of inserting into the front. 
From the official docs: "The table.insert function inserts an element in a given position of an array, moving up other elements to open space." Making it seem like an O(N) operation.

Also one thing to note is that this doesn't perfectly solve the issue of getting old responses. It seems to me that the stream_id will be released after a timeout, allowing another request to pick up that same stream_id and connection, resulting in the new request receiving the old response.
It's definitely a small probability of happening but the same can be said of a counter.

If Lua can insert in O(1) time to the front, then there shouldn't be a problem. But if it doesn't, it might be useful to think of implementing it using the counter, which is better than the current version (stream_id = 0) but not as robust as the list version.